hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAzureBlobFileSystemAuthorization.java

```diff
@@ -36,331 +40,485 @@
+  @Rule
+  public TestName name = new TestName();
 
   public ITestAzureBlobFileSystemAuthorization() throws Exception {
   }
 
   @Override
   public void setup() throws Exception {
-    this.getConfiguration().set(ConfigurationKeys.ABFS_EXTERNAL_AUTHORIZATION_CLASS, TEST_AUTHZ_CLASS);
+    boolean isHNSEnabled = this.getConfiguration().getBoolean(
+        TestConfigurationKeys.FS_AZURE_TEST_NAMESPACE_ENABLED_ACCOUNT, false);
+    Assume.assumeTrue(isHNSEnabled == true);
+    this.getConfiguration().setAbfsAuthorizerClass(TEST_AUTHZ_CLASS);
+    loadAuthorizer();
     super.setup();
   }
 
   @Test
   public void testOpenFileWithInvalidPath() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    intercept(IllegalArgumentException.class,
-        ()-> {
-          fs.open(new Path("")).close();
+    intercept(IllegalArgumentException.class, () -> {
+      fs.open(new Path("")).close();
     });
   }
 
   @Test
   public void testOpenFileAuthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_THEN_READ_ONLY_PATH).close();
-    fs.open(TEST_WRITE_THEN_READ_ONLY_PATH).close();
+    Path testFilePath = new Path(
+        TEST_WRITE_THEN_READ_ONLY_PATH_PREFIX + name.getMethodName());
+    ((MockAbfsAuthorizer) fs.getAbfsStore().getAuthorizer())
+        .setwriteThenReadOnly(testFilePath.getName(), WriteReadMode.WRITE_MODE);
+    fs.create(testFilePath).close();
+    ((MockAbfsAuthorizer) fs.getAbfsStore().getAuthorizer())
+        .setwriteThenReadOnly(testFilePath.getName(), WriteReadMode.READ_MODE);
+    fs.open(testFilePath).close();
   }
 
   @Test
   public void testOpenFileUnauthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_ONLY_FILE_PATH_0).close();
-    intercept(AbfsAuthorizationException.class,
-        ()-> {
-          fs.open(TEST_WRITE_ONLY_FILE_PATH_0).close();
+    Path testFilePath = new Path(
+        TEST_WRITE_ONLY_FILE_PATH_PREFIX_0 + name.getMethodName());
+    fs.create(testFilePath).close();
+    intercept(AbfsAuthorizationException.class, () -> {
+      fs.open(testFilePath).close();
     });
   }
 
   @Test
   public void testCreateFileAuthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    fs.create(TEST_WRITE_ONLY_FILE_PATH_0).close();
+    Path testFilePath = new Path(
+        TEST_WRITE_ONLY_FILE_PATH_PREFIX_0 + name.getMethodName());
+    fs.create(testFilePath).close();
   }
 
   @Test
   public void testCreateFileUnauthorized() throws Exception {
     final AzureBlobFileSystem fs = this.getFileSystem();
-    intercept(AbfsAuthorizationException.class,
-        ()-> {
-          fs.create(TEST_READ_ONLY_FILE_PATH_0).close();
+    Path testFilePath = new Path(
+        TEST_READ_ONLY_FILE_PATH_PREFIX_0 + name.getMethodName());
+    intercept(AbfsAuthorizationException.class, () -> {
```

Comment:
might be good to see when the check fails: in the create or in the close?

Reply:
Its create, removed the additional close call.